File: Algorithms/python/206-ReverseLinkedList/ReverseLinkedList.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        node_ret = None

        while head is not None:
            node = head
            head = head.next

            node.next = node_ret
            node_ret = node

        return node_ret

File: Algorithms/cpp/ReverseLinkedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* ptr = head;
        ListNode* ans = NULL;
        ListNode* temp = NULL;
        while (ptr) {
            temp = new ListNode(ptr->val);
            temp->next = ans;
            ans = temp;
            ptr = ptr->next;
        }
        return ans;
    }
};
